lib/puppet/provider/zabbix_proxy/ruby.rb

```diff
@@ -10,6 +10,10 @@ def create
     # Set some vars
     host = @resource[:hostname]
     ipaddress = @resource[:ipaddress]
+
+    # Normally 0 is active and 1 is passive, in the API, its 5 and 6
+    proxy_mode = @resource[:mode] + 5
```

Comment:
Why the +5? And does this work because the puppet code passes in a string while the type doesn't munge it to an integer.

Reply:
The comment explains it. I am not sure why, but in the [api documentation ](https://www.zabbix.com/documentation/3.4/manual/api/reference/proxy/create) active proxy is `5` and passive is `6` where in the config it's `0` and `1`. So to make it work I simply add 5 to the value.